code/modules/clothing/masks/gasmask.dm

```diff
@@ -2,25 +2,81 @@
 	icon_state = "gas_alt"
-	clothing_flags = BLOCK_GAS_SMOKE_EFFECT | MASKINTERNALS
+	clothing_flags = BLOCK_GAS_SMOKE_EFFECT | MASKINTERNALS | GAS_FILTERING
 	flags_inv = HIDEEARS|HIDEEYES|HIDEFACE|HIDEFACIALHAIR|HIDESNOUT
 	w_class = WEIGHT_CLASS_NORMAL
 	inhand_icon_state = "gas_alt"
 	gas_transfer_coefficient = 0.01
 	permeability_coefficient = 0.01
 	flags_cover = MASKCOVERSEYES | MASKCOVERSMOUTH | PEPPERPROOF
 	resistance_flags = NONE
+	tint = 0.9
+	///Max numbers of installable filters
+	var/max_filters = 1
+	///List to keep track of each filter
+	var/list/gas_filters
+
+/obj/item/clothing/mask/gas/examine(mob/user)
+	. = ..()
+	if(max_filters > 0)
+		. += "<span class='notice'>The [src] has [max_filters] slot\s for filters are [max_filters].</span>"
+	if(LAZYLEN(gas_filters) > 0)
+		. += "<span class='notice'>Currently there are [LAZYLEN(gas_filters)] filter\s with [get_filter_durability()]% durability.</span>"
+
+/obj/item/clothing/mask/gas/attackby(obj/item/filter, mob/user)
+	. = ..()
+	if(!istype(filter, /obj/item/gas_filter))
+		return TRUE
+	if(LAZYLEN(gas_filters) >= max_filters)
+		return TRUE
+	if(!user.transferItemToLoc(filter, src))
+		return TRUE
+	LAZYADD(gas_filters, filter)
+	if(LAZYLEN(gas_filters) > 0)
+		has_filter = TRUE
+
+///Check _masks.dm for this one
+/obj/item/clothing/mask/gas/consume_filter(datum/gas_mixture/breath)
+	if(LAZYLEN(gas_filters) <= 0 || max_filters == 0)
+		return breath
+	var/obj/item/gas_filter/gas_filter = pick(gas_filters)
+	var/datum/gas_mixture/filtered_breath = gas_filter.reduce_filter_status(breath)
+	if(gas_filter.filter_status <= 0)
+		LAZYREMOVE(gas_filters, gas_filter)
+		qdel(gas_filter)
+	if(LAZYLEN(gas_filters) <= 0)
+		has_filter = FALSE
+	return filtered_breath
+
+/**
+ * Getter for overall filter durability, takes into consideration all filters filter_status
+ */
+/obj/item/clothing/mask/gas/proc/get_filter_durability()
+	var/max_filters_durability = LAZYLEN(gas_filters) * 100
+	var/current_filters_durability
+	for(var/obj/item/gas_filter/gas_filter as anything in gas_filters)
+		current_filters_durability += gas_filter.filter_status
+	var/durability = (current_filters_durability / max_filters_durability) * 100
+	return durability
 
 /obj/item/clothing/mask/gas/atmos
 	name = "atmospheric gas mask"
-	desc = "Improved gas mask utilized by atmospheric technicians. Still not very good at blocking gas flow, but it's flameproof!"
+	desc = "Improved gas mask utilized by atmospheric technicians. It's flameproof!"
 	icon_state = "gas_atmos"
 	inhand_icon_state = "gas_atmos"
 	armor = list(MELEE = 0, BULLET = 0, LASER = 0,ENERGY = 0, BOMB = 0, BIO = 0, RAD = 10, FIRE = 20, ACID = 10)
 	w_class = WEIGHT_CLASS_SMALL
 	gas_transfer_coefficient = 0.001 //cargo cult time, this var does nothing but just in case someone actually makes it do something
 	permeability_coefficient = 0.001
 	resistance_flags = FIRE_PROOF
+	max_filters = 3
+
+/obj/item/clothing/mask/gas/atmos/Initialize()
```

Comment:
why do atmos gas masks spawn with filters, but not normal ones?

like, why would NT put a gas mask in a fire closet, but not a gas filter to go with it? it seems like that'd be cheaper and easier than putting a gas filter in everyone's roundstart internals boxes (which are all damaged, for some reason)

Reply:
Ah I was going to Change this to give every gas mask an initial filter